### PR TITLE
fix: close menu after duplication

### PR DESCRIPTION
**Pull request by JayChen2004**
# Description

Fixed an issue where menu would not close after duplicating in journey editor

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/35391220/todos/6948516862)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] dropdown tab should close after duplicating

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/EditToolbar/DuplicateBlock/DuplicateBlock.tsx [hunk 1]
```diff
@@ -32,6 +33,7 @@ export const BLOCK_DUPLICATE = gql`
 
 export function DuplicateBlock({
   variant,
+  closeMenu,
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/EditToolbar/DuplicateBlock/DuplicateBlock.tsx:36]: Add a test in this component that the `closeMenu` function gets called

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.spec.tsx [hunk 2]
```diff
@@ -313,6 +314,67 @@ describe('EditToolbar Menu', () => {
...
+            </EditorProvider>
+          </JourneyProvider>
+        </MockedProvider>
+      </SnackbarProvider>
+    )
+    expect(queryByRole('menu')).not.toBeInTheDocument()
+    fireEvent.click(getByTestId('MoreIcon'))
+    await waitFor(() => expect(queryByRole('menu')).toBeInTheDocument())
+    fireEvent.click(getByTestId('JourneysAdminMenuItemDuplicate-Block'))
+    await waitFor(() => expect(queryByRole('menu')).not.toBeInTheDocument())
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.spec.tsx:375]: You're only asserting that the menu is no longer there, check that the result of you're mock is called.

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.spec.tsx [hunk 3]
```diff
@@ -313,6 +314,68 @@ describe('EditToolbar Menu', () => {
     expect(getByRole('menuitem', { name: 'Delete Card' })).toBeInTheDocument()
   })
 
+  it('should close the menu upon clicking duplicating from menu', async () => {
+    const selectedBlock: TreeBlock<TypographyBlock> = {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.spec.tsx:318]: This test is looking good! Can you do a refactor of the mocking so it matches a similar style to how its done here? https://github.com/JesusFilm/core/blob/f89a4172f4961d7e6b5e9e8faec28cad3a329c65/apps/journeys-admin/src/components/AccessDenied/AccessDenied.spec.tsx

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Editor/EditToolbar/DuplicateBlock/DuplicateBlock.spec.tsx [hunk 4]
```diff
@@ -238,6 +238,51 @@ describe('DuplicateBlock', () => {
...
+              variables: {
+                id: step.id,
+                journeyId: 'journeyId',
+                parentOrder: null
+              }
+            },
+            result
+          }
+        ]}
+      >
```
[the remark this answers is not in the document]
Comment (JayChen2004) [reply apps/journeys-admin/src/components/Editor/EditToolbar/DuplicateBlock/DuplicateBlock.spec.tsx:268]: my handleClick function is inside a function handleDuplicateBlock that needs this mock

File: apps/journeys-admin/src/components/Editor/EditToolbar/DuplicateBlock/DuplicateBlock.spec.tsx [hunk 5]
```diff
@@ -238,6 +238,51 @@ describe('DuplicateBlock', () => {
...
 
+  it('should call closeMenu after clicking duplicate', async () => {
+    const result = jest.fn(() => ({
+      data: {
+        blockDuplicate: {
+          id: 'duplicatedId',
+          parentOrder: 1
+        }
+      }
+    }))
```
[the remark this answers is not in the document]
Comment (JayChen2004) [reply apps/journeys-admin/src/components/Editor/EditToolbar/DuplicateBlock/DuplicateBlock.spec.tsx:249]: this is needed for handleClick to be called too

File: apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.spec.tsx [hunk 6]
```diff
@@ -313,6 +314,66 @@ describe('EditToolbar Menu', () => {
...
+      parentBlockId: 'card1.id',
+      parentOrder: 0,
+      content: 'Title',
+      variant: null,
+      color: null,
+      align: null,
+      children: []
+    }
+
+    const mockBlockDuplicate: MockedResponse = {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.spec.tsx:330]:
```suggestion
    const mockBlockDuplicate: MockedResponse<BlockDuplicate> = {
```

**Review by csiyang — CHANGES_REQUESTED** [review]

**Review by csiyang — APPROVED** [review]

**Pull request merged**